### PR TITLE
API keys, enforce login everywhere, improve test coverage

### PR DESCRIPTION
**Pull request by davea**
- Ensures users must log in before accessing any page
- API calls can be made by logged-out users if they include an `api_key` request parameter
- Extend test coverage for UPRN lookup
- Smaller improvements e.g. branding, health check URL, fix OSM maps in admin

### CODE REVIEW

File: labour_project/settings.py [hunk 1]
```diff
@@ -153,6 +153,7 @@
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'login_required.middleware.LoginRequiredMiddleware',
```
Comment (dracos) [inline labour_project/settings.py]: This commit is missing the adding of the new package, I think? Or was it already added without the code.

File: mapit_labour/templates/mapit_labour/uprn.html [hunk 2]
```diff
@@ -11,8 +11,8 @@ <h2>UPRN: {{ uprn.uprn }}</h2>
       <li>Postcode: <a href="{% url "mapit-postcode" postcode=uprn.postcode format="html" %}">{{ uprn.postcode }}</a>
       <li>OSGB E/N: {{ uprn.easting }}, {{ uprn.northing }}
       <li>{% trans "WGS84 lat/lon" %}: <a href="https://tools.wmflabs.org/geohack/geohack.php?params={{ uprn.wgs84_lat|floatformat:-6 }};{{ uprn.wgs84_lon|floatformat:-6 }}">{{ uprn.wgs84_lat|floatformat:-6 }}, {{ uprn.wgs84_lon|floatformat:-6 }}</a>
-      {% url json_view uprn=uprn.uprn as json_url %}
-      <li>{% blocktrans %}Get <a href="{{ json_url }}">this data as JSON</a>{% endblocktrans %}</li>
+      {% url json_view uprn=uprn.uprn format='json' as json_url %}
```
Comment (dracos) [inline mapit_labour/templates/mapit_labour/uprn.html:14]: Just wondered why you added this (no issue with it)

File: bin/mapit_make_css [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#
+# mapit_make_css:
+# Generate CSS files from SCSS files.
+# The default MapIt mapit.scss file assumes that the Python package has been
+# installed in editable mode and that the mysociety-docs-theme git submodule
+# has been checked out. This isn't the case when MapIt has been installed by
+# Poetry, so we need to replace the mapit.scss with our own that uses the
+# correct path to mysociety-docs-theme (which is a git submodule of the
+# labour-mapit project and *is* present).
+# This whole thing should probably be replaced by django-compressor and
```
Comment (dracos) [inline bin/mapit_make_css:11]: Sorry this was a bit fiddly, yeah, can probably be improved nowadays.

File: mapit_labour/tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from django.test import TestCase
```
Comment (dracos) [inline mapit_labour/tests/test_models.py:1]: Can squash this in with the Add API keys commit presumably

File: labour_project/settings.py [hunk 5]
```diff
@@ -252,7 +252,7 @@
 ]
 
 API_KEY_AUTH_ALLOWED_PATHS = [
-    r'^/uprn/\d+\.json',
+    r'^/uprn/\d+(\.json)?$',
```
Comment (dracos) [inline labour_project/settings.py:256]: IF you had time, could rebase to have these fixes within the previous places, but no biggie.

File: mapit_labour/middleware.py [hunk 6]
```diff
@@ -1,19 +1,30 @@
...
 
 class LoginOrAPIKeyRequiredMiddleware(LoginRequiredMiddleware):
-    def _api_key_exists(self, request):
-        if api_key := request.GET.get("api_key"):
-            return APIKey.objects.filter(key=api_key).exists()
+    def _api_key_exists_and_allowed(self, request):
+        if not any(url.match(request.path) for url in ALLOWED_PATHS):
+            return False
+
+        if APIKey.objects.filter(
```
Comment (dracos) [inline mapit_labour/middleware.py:22]: Stick the test for authenticated and returning False here in this commit, makes sense as part of that, is a bug fix

**Review by dracos — APPROVED** [review]
Looks fine, could probably tidy up a little for future tidiness but that's all I think :)

File: labour_project/settings.py (hunk 1 above)
Comment (davea) [reply labour_project/settings.py]: Yep, it got added earlier in 90a46f460eab56e0985428720e57a611cfbe5366.

File: mapit_labour/templates/mapit_labour/uprn.html (hunk 2 above)
Comment (davea) [reply mapit_labour/templates/mapit_labour/uprn.html:14]: The `format='json'` part? I'm afraid I can't remember :/

**Comment by davea** [comment]
Merged into main as f656aace3a61517af2f9c9faba4758f2ecb01490.

**Pull request closed without merging**